cmd/server/server.go

```diff
@@ -0,0 +1,153 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"net/http"
+	"time"
+
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/freeverseio/laos-universal-node/internal/platform/blockchain"
+	internalRpc "github.com/freeverseio/laos-universal-node/internal/rpc"
+)
+
+type HTTPServerer interface {
+	ListenAndServe() error
+	Shutdown(context.Context) error
+	SetKeepAlivesEnabled(bool)
+	SetAddr(string)
+	SetHandler(http.Handler)
+}
+
+// The real implementation that wraps http.Server
+type HTTPServer struct {
+	server *http.Server
+}
+
+func (h *HTTPServer) ListenAndServe() error {
+	return h.server.ListenAndServe()
+}
+
+func (h *HTTPServer) Shutdown(ctx context.Context) error {
+	return h.server.Shutdown(ctx)
+}
+
+func (h *HTTPServer) SetKeepAlivesEnabled(v bool) {
+	h.server.SetKeepAlivesEnabled(v)
+}
+
+func (h *HTTPServer) SetAddr(addr string) {
+	h.server.Addr = addr
+}
+
+func (h *HTTPServer) SetHandler(handler http.Handler) {
+	h.server.Handler = handler
+}
+
+type RPCServerer interface {
+	RegisterName(name string, receiver interface{}) error
+	ServeHTTP(http.ResponseWriter, *http.Request)
+}
+
+type Server struct {
+	RPCServer  RPCServerer
+	HTTPServer HTTPServerer
+}
+
+type ServerOption func(*Server) error
+
+// WithEthService initializes and registers the eth service with the server.
+func WithEthService(ethcli blockchain.EthClient) ServerOption {
+	return func(s *Server) error {
+		eth := internalRpc.NewEthService(ethcli.Client())
+		return s.RPCServer.RegisterName("eth", eth)
+	}
+}
+
+// WithNetService initializes and registers the net service with the server.
+func WithNetService(chainID uint64) ServerOption {
+	return func(s *Server) error {
+		net := internalRpc.NewNetService(chainID)
+		return s.RPCServer.RegisterName("net", net)
+	}
+}
+
+// WithSystemHealthService initializes and registers the system health service with the server.
+func WithSystemHealthService() ServerOption {
+	return func(s *Server) error {
+		systemHealth := internalRpc.NewSystemHealthService()
+		return s.RPCServer.RegisterName("health", systemHealth)
+	}
+}
+
+// WithRPCServer allows you to provide a custom RPCServerer implementation.
+func WithRpcServer(rpcServer RPCServerer) ServerOption {
+	return func(s *Server) error {
+		s.RPCServer = rpcServer
+		return nil
+	}
+}
+
+// WithHTTPServer allows you to provide a custom HTTPServerer implementation.
+func WithHTTPServer(httpServer HTTPServerer) ServerOption {
+	return func(s *Server) error {
+		s.HTTPServer = httpServer
+		return nil
+	}
+}
+
+func NewServer(opts ...ServerOption) (*Server, error) {
```

Comment:
Rename this to `New` so when external packages call this method they will do: `server.New`

Reply:
I don't like it. I prefer NewServer. This is much clearer (in case we decide to rename or change the pkg)